sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/_shared/_polling.py

```diff
@@ -0,0 +1,204 @@
+        # type: (Optional[int]) -> Model
+        """Returns the result of the long running operation without blocking until its completion.
+
+        :returns: The deserialized resource of the long running operation,
+         if one is available.
+        :raises ~azure.core.exceptions.HttpResponseError: Server problem with the query.
+        """
+        return self._polling_method.resource()
+
+    @distributed_trace
+    def wait(self, timeout=None):
+        # type: (Optional[int]) -> None
+        """Wait on the long running operation for a specified length
+        of time. You can check if this call as ended with timeout with the
+        "done()" method.
+
+        :param int timeout: Period of time to wait for the long running
+         operation to complete (in seconds).
+        :raises ~azure.core.exceptions.HttpResponseError: Server problem with the query.
+        """
+
+        if not self._polling_method.finished():
+            self._done = threading.Event()
+            self._thread = threading.Thread(
+                target=with_current_context(self._start),
+                name="KeyVaultOperationPoller({})".format(uuid.uuid4()))
+            self._thread.daemon = True
+            self._thread.start()
+
+        if self._thread is None:
+            return
+        self._thread.join(timeout=timeout)
+        try:
+            # Let's handle possible None in forgiveness here
+            raise self._exception  # type: ignore
+        except TypeError: # Was None
+            pass
+
+
+
+class DeleteResourcePoller(PollingMethod):
+    def __init__(self, interval=2):
+        self._command = None
+        self._deleted_resource = None
+        self._polling_interval = interval
+        self._status = "deleting"
+
+    def _update_status(self):
+        # type: () -> None
+        try:
+            self._command()
+            self._status = "deleted"
+        except ResourceNotFoundError:
+            self._status = "deleting"
+        except HttpResponseError as e:
+            if e.status_code == 403:
+                self._status = "deleted"
+            else:
+                raise
+
+    def initialize(self, client, initial_response, _):
+        # type: (Any, Any, Callable) -> None
+        self._command = client
+        self._deleted_resource = initial_response
+
+    def run(self):
+        # type: () -> None
+        try:
+            while not self.finished():
+                self._update_status()
+                time.sleep(self._polling_interval)
+        except Exception as e:
+            logger.warning(str(e))
+            raise
+
+    def finished(self):
+        # type: () -> bool
+        return not self._deleted_resource.recovery_id or self._status == "deleted"
+
+    def resource(self):
+        # type: () -> Any
+        return self._deleted_resource
+
+    def status(self):
+        # type: () -> str
+        return self._status
+
+class RecoverDeletedResourcePoller(PollingMethod):
```

Comment:
Looks like the only difference between these `PollingMethod` implementations is the string indicating completion. Could you factor out a single class with an `__init__` parameter for that?

Reply:
Made DeletePollingMethod inherit from this, thanks!